bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -1690,6 +1692,12 @@ private ImageHandle init(ImageData i, int zoom) {
 	}
 }
 
+private void assertCondition(boolean condition, String message) {
+    if (condition && strictChecks) {
+        System.err.print(message + "\nSee https://github.com/eclipse-platform/eclipse.platform.swt/tree/master/examples/org.eclipse.swt.snippets/src/org/eclipse/swt/snippets/Snippet384.java for supported variants.");
```

Comment:
I would also add more general statement than only referring to an example snippet here. Something like "Images with custom parts should be created with an ImageGcDrawer", as that's the actualy replacement for any kind of image whose creation involved a GC by now.

Reply:
Do we already have a N&N for `ImageGcDrawer` (I still find the name quite confusing) we can reference here? Such N&N can then also reference snippets and whatever seems useful in that area.)

Reply:
https://eclipse.dev/eclipse/news/news.html?file=4.35/platform_isv.html